src/main/java/com/laytonsmith/core/events/drivers/BlockEvents.java

```diff
@@ -1519,4 +1519,75 @@ public boolean modifyEvent(String key, Mixed value, BindableEvent event) {
 			return false;
 		}
 	}
+
+	@api
+	public static class block_form extends AbstractEvent {
+
+		@Override
+		public String getName() {
+			return "block_form";
+		}
+
+		@Override
+		public String docs() {
+			return "{newblock: <string match> Type of new block state}"
+					+ "Called when a block is formed or spreads based on world conditions."
+					+ " Cancelling this event block will not be formed."
```

Comment:
`Cancelling this event will cause the block to not be formed.`

Reply:
Fixed